addons/im_livechat/controllers/main.py

```diff
@@ -78,15 +78,37 @@ def _get_guest_name(self):
 
     @http.route('/im_livechat/get_session', methods=["POST"], type="jsonrpc", auth='public')
     @add_guest_to_context
-    def get_session(self, channel_id, anonymous_name, previous_operator_id=None, chatbot_script_id=None, persisted=True):
+    def get_session(self, channel_id, previous_operator_id=None, chatbot_script_id=None, persisted=True):
         store = Store()
-        user_id = None
         country_id = None
         channel = request.env["discuss.channel"]
         guest = request.env["mail.guest"]
-        # if the user is identifiy (eg: portal user on the frontend), don't use the anonymous name. The user will be added to session.
+        chatbot_script = (
+            request.env["chatbot.script"]
+            .sudo()
+            .with_context(lang=request.env["chatbot.script"]._get_chatbot_language())
+            .browse(chatbot_script_id)
+        )
+        livechat_channel = (
+            request.env["im_livechat.channel"].with_context(lang=False).sudo().browse(channel_id)
+        )
+        if not livechat_channel:
+            return False
+        operator = False
```

Comment:
It's generally better to initialize to empty recordsets. This way, long ternary below can just become
(agent.XXX or bot.XXX)